Add support for compilation on Windows using VS or MinGW

smlu opened the pull request.

smlu committed:
-Updated .gitignore to ignore VisualStudio

smlu committed:
Add Visual Studio project

smlu committed:
Update code to allow compiling on Windows

smlu committed:
link to static zlib

smlu committed:
Update linking to libevent

BenWiederhake commented:
There already is a effort underway to make this run on Windows. Why do you propose a second one?

A few things I've noticed:

- Typo in the first commit
- PR is based against master
- You're linking against OpenSSL in your VS project files. This calls for trouble, due to the licensing issues. The older Windows port already solved this, since there's a `--disable-openssl` switch in the test branch.
- The VS xml files (9df0d7b) are very intrusive. Isn't there a smaller build system overhead?
- Your config.bat looks auto-generated. Add source and instructions how to re-generate.
- `_PRINTF_INT64_` is not a portable way of doing it. The other PR already solves this portably.
- There are a lot of modifications and parallel, nearly duplicated code sections. The existing approach also does this, but isn't as bad.
- Solving portability problems by simply adding a few `(int)` casts is great if you only run it on your own machine. However, this code will run on lots of machines.

In essence: Thanks for trying to make tgl Windows-ready! But it would have been much more effective if you work together with the existing efforts.

smlu commented:
The previous pull request #62 does not work under msvc, so this proposal also adds support for Microsoft C++ compiler. Maybe I should have made it more clear.
It's based on master branch because our project requires a stable version.
If there is a request for Windows/MSVC support in general I am willing to re-base it to test branch.

To answer your other inquiries:
- Sorry for the typos there.
- I did not know there was a licensing issue with OpenSSL but master branch does not work without it. This will be fixed if moved to test branch.
- Vs xml, the usual practice with visual studio is to go with the project file. However, there is msvc nmake, a Unix kind of make tool but it does not support unix makefile.  Maybe it could be ported somehow, I've never tried it           .
      Suggestion: it would be easier if tgl supported cmake-like cross platform project generator.
- config.bat was not auto-generated. I based it on what configure and makefile do.
- _PRINTF_INT64_ is done the same way as previous pr #62
- The code in general was not modified at all. There are more additions to the code rather than changes of the original code. MSVC unfortunately does not support
  much of the Unix/Ansi C API functions or those functions have been deprecated. There are also cases where equal functions require additional parameters under winapi. That is why there are so many added preproc wrappers that solves this problem and it is hard to do it any other way.
- Sorry for the casts. They should not be there, will fix it.

smlu committed:
Add preprocessor macro _CRT_NONSTDC_NO_DEPRECATE,
Remove macros strdup, read, write, close, lseek

smlu committed:
Fix call to the  PEM_read_RSAPublicKey on windows when using DLL version of OpenSSL ##
Fix memory leak

smlu committed:
Update tgl_pad_rsa_encrypt && tgl_pad_rsa_decrypt

smlu committed:
Fix Memory leaks

smlu committed:
Update add_log_event; add log entry when writing to binlog file is not successful

smlu committed:
Fix pfs bug when TLS->enable_pfs is false ##
Add tglq_regen_queries_from_old_session ##
Update work_new_session_created ##
Fix tglmp_on_start to not exit if error occurs

smlu committed:
Add on_failed_login to tgl_update_callback struct ##
Update tgl_int

smlu committed:
Update updates.h && updates.c

smlu committed:
Add comment

Commit:
Add Makefile.android && Update code to compile on unix system

Commit:
Update Readme.md

smlu committed:
Update README.md

Commit:
Update Makefile.android

Commit:
Merge branch 'master' of https://github.com/ZeroPass/tgl

smlu committed:
Fix unknow ptr size error on windows
